llvm/test/Feature/Repo/repo_pruning_dependents.ll

```diff
@@ -1,27 +1,47 @@
-; This testcase tests the program repository pruning pass.
+; This testcase tests the program repository pruning pass for the dependents.
 ;
-; If function 'A' is dependent on function 'B' and 'B' is dependent
-; on function 'C', both TickeNodes of 'B' and 'C' need to be added
-; into in the 'repo.tickets' during the pruning.
+; In the llvm-project-prepo project, the GO's digest could be generated either by the
+; RepoHashGenerator pass or by the RepoObjectWriter. When a new global object (GO) is
+; generated by an LLVM optimisation pass after the RepoHashGenerator pass, the digest
+; will be generated by the RepoObjectWriter. If a fragment has an external fixup
+; referencing this GO, we add the GO's compilation member address to the dependents.
+;
+; When a fragment is pruned, all its dependents need to be recorded in the repo.tickets
+; metadata to guarantee that the dependents (those GO generated by later optimizations)
+; are present in the final compilation record.
+;
+; In this test, the digest of 'A' is given in the TicketNode. The digests of `B` and `C`
+; will be generated by the the RepoObjectWriter. Since Function 'A' is dependent on
+; function 'B' and 'B' is dependent on function 'C', both TickeNodes of 'B' and 'C' need
+; to be added into the 'repo.tickets' during the pruning.
 
 ; The testcase includes three steps:
 ; Step 1: Build the code and create the database 'clang.db' which contains all Tickets.
+;         In this step, the digest of function `B` and `C` are generated by the RepoObjectWriter
+;         and added these two functions into the dependents of the function 'A'.
 ; Step 2: Dump all compilation members in the database;
+;         There are 3 compilation members, which are functions `A`, `B` and `C`.
 ; Step 3: Re-build the same IR code again.
+;         In this step, the IR code firstly pass the RepoHashGenerator pass to generate the
+;         digests of functions 'A', 'B' and 'C'. During the RepoPruning pass, the function 'A'
+;         is pruned and its dependents 'B' and 'C' are added.
 ; Step 4: Dump all compilation members in the database again;
+;         The compilations, which are generated by step 1 and step 3, should be the same.
+;         Therefore, there are only one compilation, which inludes three members: functions
+;          `A`, `B` and `C`.
 ; Step 5: Check that the step 2 and step 4 generate the same compilation members.
-
-; This test only works for the Debug build because the digest of A is calculated for the Debug build.
-
+;
+; The test need to be updated if the compiler optimisation passes change.
```

Comment:
To be clear, does this need to be updated if a pass itself changes, or just if the pipeline of optimisation passes changes?

Reply:
I have updated the comment to `The test needs to be updated if either the name of the optimisation passes or the pipeline of optimisation passes changes.`